examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
+    base_model = resnet_v2.ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(352, 608, 3)
+    )
+    d0 = base_model.get_layer("input_1").output
+    d1 = base_model.get_layer("conv1_conv").output
+    d2 = base_model.get_layer("conv2_block3_1_relu").output
+    d3 = base_model.get_layer("conv3_block4_1_relu").output
+
+    decode1 = upscale_unit(previous_input=d3, current_input=d2, target_channels=128)
+    decode2 = upscale_unit(previous_input=decode1, current_input=d1, target_channels=64)
+    decode3 = upscale_unit(previous_input=decode2, current_input=d0, target_channels=32)
+
+    out_conv = keras.layers.Conv2D(3, 5, padding="same", activation="sigmoid")(decode3)
+    denoiser_model = keras.Model(base_model.input, out_conv, name="denoiser")
+    return denoiser_model
+
+
+"""
+## Training
+"""
+
+denoiser_net = denoiser()
+optimizer = tf.keras.optimizers.Nadam(1e-4)
+denoiser_net.compile(optimizer=optimizer, loss="mse", metrics=["mae"])
+
+early_stop = callbacks.EarlyStopping(patience=1)
+ckpt = callbacks.ModelCheckpoint(
+    "best_ckpt.h5", save_best_only=True, save_weights_only=True, verbose=1
+)
+
+denoiser_net.fit(
+    train_pipe,
+    validation_data=val_pipe,
+    batch_size=batch_size,
+    epochs=1,
+    steps_per_epoch=train_step,
+    validation_steps=val_step,
+    workers=-1,
+    callbacks=[early_stop, ckpt],
+)
+
+"""
+## Evaluation Utilities
+
+Though Image Super-Resolution or Denoiser models can be evaluated with PSNR, SSIM metrics
+for perceptual quality assessments our task of text restoration requires a bit more, our
+objective is to enhance simple OCR accuracy engine where our model can act as a
+pre-processor, to serve the purpose we will use <a
+href="https://towardsdatascience.com/evaluating-ocr-output-quality-with-character-error-ra
+te-cer-and-word-error-rate-wer-853175297510">WER(Word Error Rate) </a> as a metric to
+evaluate the real-world performance. The following codes will help plot the visual
+comparisons of outputs given the raw noisy input and model restored input to tesseract
+OCR engine.
+"""
```

Comment:
@anish9 rewrite these statements with a more general and simpler form. Make compound sentences into simple sentences.  For example, the first sentence is not convenient to read.